### ISSUE TITLE
Make Alt+F forward a word again

### ISSUE THREAD

**Issue by qorg11** [profanity-im/profanity#1580]
## Expected Behavior

The input moves to the next word (Like in emacs and most shells)
## Current Behavior

It sets the attention flag
## Possible Solution

Configure keybindings
## Steps to Reproduce (for bugs)

1. Press Alt + F while writing a message with more than 2 words

**Comment by jubalh** [comment]
I totally forgot about usual readline shortcuts since I never use them.
We should make sure not to overwrite the most default ones (that make sense) probably: https://readline.kablamo.org/emacs.html

**Comment by jubalh** [comment]
So what would be good alternatives for the key combos?

alt-f flag window as marked
alt-m - switch between marked windows

alt-a marking as unread

I'm tempted to just let them stay as they are since I guess all the readline combos can also be done with ctrl+arrow keys. And we probably want short key combos for the marking/switching between windows.

**Comment by qorg11** [comment]
Well, if it will be less pain for you, leave them like that. Altough i'd like it to be possible to use Alt+f again

**Comment by jubalh** [comment]
It's not hard to change to code to use another binding.
We just need to have a good idea with letters that make sense and at the same time not use default readline letters :/

Do you have sugestions?

**Comment by qorg11** [comment]
Hm, well I think Alt-n doesn't make sense but it is near m so you can quickly press Alt-n and Alt-m?

**Comment by jubalh** [comment]
What about the others though?

**Comment by 4drift** [comment]
I agree with alt+n as it is convenient. I am greatly displeased by having alt+f gone as I use it to jump words ever constantly and ctrl+arrows doesn't even work for me.

Besides it is terribly inconvenient compared to alt+f/b -- I use it in the terminal constantly too. I'd think more people would have a problem with this binding being taken over. Having a memorable letter just isn't reason enough to overwrite a default terminal binding.

**Comment by jubalh** [comment]
> I am greatly displeased by having alt+f gone as I use it to jump words ever constantly

We don't need to know about this. The original poster said this already.

> and ctrl+arrows doesn't even work for me.

Then fix your setup.

>  Having a memorable letter just isn't reason enough to overwrite a default terminal binding.

alt+f is not a "default terminal binding". Actually it has nothing to do with your terminal. It's the default binding for GNU Readline in emacs mode.
If we would have not chosen to use the readline library then this terminal wouldn't work out of the box at all. Like is the case for many applications out there.

Now with this being clarified I would again ask for alternative keybinding suggestions. The change is not big, just one/two lines that need to be changed. But someone will have to come up with an alternative binding.

**Comment by allora** [comment]
> We don't need to know about this. The original poster said this already.

TBH its useful to know how many folks use some (seemingly very common and probably default in some cases) bindings. I Also would prefer profanity to respect this specific binding (atl+f/b).

> Then fix your setup.

TBH, not every terminal out there will do ctrl+arrow keys out of the box. Rxvt-Unicode (urxvt) or UXTerm for example. So its not really "fix your setup" since its not always a -default- on every terminal emulator out there. Additionally arrow keys are highly inconvenient.

> alt+f is not a "default terminal binding". Actually it has nothing to do with your terminal. It's the default binding for GNU Readline in emacs mode.

No need to be pedantic :P I think we get the gist of the comment.

> Now with this being clarified I would again ask for alternative keybinding suggestions. The change is not big, just one/two lines that need to be changed. But someone will have to come up with an alternative binding.

I think you missed the part where he said alt+n is convenient. It was at the beginning. And I will second that suggestion.

On the flip side, down the road a better solution to trying to avoid conflicting bindings in who knows how many different terminal emulators people use, would be to let folks customize the bindings them selves. But that is def a bigger task than just changing some hard coded values :)

**Comment by jubalh** [comment]
So this is what's going to happen:
Someone will go through the list of keybindings in profanity and through the list of default keybindings from readline (mentioned in https://github.com/profanity-im/profanity/issues/1580#issuecomment-881247688) and will propose the changes here with explanation why they make sense. If noone can do it, then it will stay like it is.

**Comment by jubalh** [comment]
We only need to change alt+f all the rest is okay? Did someone check?
We could use alt+v instead of alt+f.
Reasons:
* v is close enough on keyboard to the f 
* f and v sound similar so we could still say "vlag" ;)
* v seems free?

**Comment by cockroach** [comment]
v sounds like a good alternative to me

**Comment by 4drift** [comment]
Some alt bindings are not implemented: alt+c, alt+l, and alt+t it seems. I'm not sure where I got this list now, but these were once given as a reference for the main bindings to me. 

Alt + BackSpace to delete to a separator/delimitor!
Ctrl + a – go to the start of the command line
Ctrl + e – go to the end of the command line
Ctrl + k – delete from cursor to the end of the command line
Ctrl + u – delete from cursor to the start of the command line
Ctrl + w – delete from cursor to start of word (i.e. delete backwards one word)
Ctrl + y – paste word or text that was cut using one of the deletion shortcuts (such as the one above) after the cursor
Ctrl + xx – move between start of command line and current cursor position (and back again)
Alt + b – move backward one word (or go to start of word the cursor is currently on)
Alt + f – move forward one word (or go to end of word the cursor is currently on)
Alt + d – delete to end of word starting at cursor (whole word if cursor is at the beginning of word)
Alt + c – capitalize to end of word starting at cursor (whole word if cursor is at the beginning of word)
Alt + u – make uppercase from cursor to end of word
Alt + l – make lowercase from cursor to end of word
Alt + t – swap current word with previous
Ctrl + f – move forward one character
Ctrl + b – move backward one character
Ctrl + d – delete character under the cursor
Ctrl + h – delete character before the cursor
Ctrl + t – swap character under cursor with the previous one

Ctrl + l – clear the screen
Ctrl + s – stops the output to the screen (for long running verbose command)
Ctrl + q – allow output to the screen (if previously stopped using command above)
Ctrl + c – terminate the command
Ctrl + z – suspend/stop the command

I don't think EVERY single readline binding has bearing on this discussion. I would think the movement ones are the ones most important?

**Comment by jubalh** [comment]
Switched alt-f to alt-v. Hope everybody is happy now :)
0.11.1 will have this.
